### ISSUE TITLE
Downloading the Image for the whole dataset

### ISSUE THREAD

**Issue by uakarsh** [furkanbiten/idl_data#2]
Hi @furkanbiten, you have given the link for downloading the Raw OCR, however I am not able to download the corresponding images (not sure about how to go for it). Can you refer me to the appropriate place?

Thanks,
Akarsh

**Comment by furkanbiten** [comment]
Hey @uakarsh,

I believe you are referring to download the original pdfs. 
You can find the information in here: https://github.com/furkanbiten/idl_data#download-raw-pdfs

You need an AWS account to download. Because of the license issues and IDL being huge, I couldn't upload to our university servers. 

Let me know if this is what you are referring to.

**Comment by furkanbiten** [comment]
Closing the issue for now, reopen it if this doesnt solve your issue.

**Comment by uakarsh** [comment]
Hi @furkanbiten, can you tell me the steps/ refer to appropriate location to download the pdfs. Here is what I am able to see.

![image](https://user-images.githubusercontent.com/55104596/206382633-5f621fdb-537a-4cb9-81fe-787d458f0f29.png)

Regards,
Akarsh

**Comment by furkanbiten** [comment]
You need to install aws cli and run the command
`aws s3 cp s3://edu.ucsf.industrydocuments.artifacts [PATH_TO_SAVE] --recursive`

**Comment by uakarsh** [comment]
Got it. I would try and let you know. 

Thanks,
Akarsh

**Comment by uakarsh** [comment]
Hi there, I tried it, and this is what I am able to see on my terminal

![image](https://user-images.githubusercontent.com/55104596/206641523-1fa3de20-f68d-4843-ac9e-ca1fa5ccf97b.png)


Something wrong in my steps? I configured the AWS account using aws cli, and then ran the command 
`aws s3 cp s3://edu.ucsf.industrydocuments.artifacts ./ --recursive`

**Comment by uakarsh** [comment]
And, just wanted to know, is this [link](https://ucsf.app.box.com/v/IDL-DataSets/file/681105722857), different from the one, you described, i.e [here](https://s3.console.aws.amazon.com/s3/buckets/edu.ucsf.industrydocuments.artifacts?region=us-west-2)

PS:

While reading your [paper](https://arxiv.org/pdf/2202.12985.pdf), on page 5, I saw that you have mentioned the dataset [link](https://s3-us-west-2.amazonaws.com/edu.ucsf.industrydocuments.artifacts/), I think it has been removed from public access.

**Comment by furkanbiten** [comment]
Hey @uakarsh,

The metadata is stored in this link: https://ucsf.app.box.com/v/IDL-DataSets/
All the information such as type of documents, year, etc. is stored in these links. So no pdfs can be found in there.

It seems like you are right. It does seem like they removed the s3 bucket from public access 😢 

However, all hope is not lost! After doing a bit of snooping, I have realized that they are using `https://download.industrydocuments.ucsf.edu` to store all the pdfs. 

So as an example: the pdf ssyk02333.pdf would be stored in `https://download.industrydocuments.ucsf.edu/s/s/y/k/ssyk0233/ssyk0233.pdf`

I am assuming a very simple script that goes over all the pdfs and download it would work! Although one must be always careful on getting the IP blocked, so treat the parallelization carefully. 
 
@rubenpt91, can we have all the pdf names stored in json so that we can put it in our repo? 
I think it would be nice to have BOTH: 
1) all the pdfs names.
2) all the pdfs names which we have the OCR for.

**Comment by furkanbiten** [comment]
NOTE TO FUTURE USERS:

If you would like to download the WHOLE IDL, you can get all the filenames from the metadata link:
https://ucsf.app.box.com/v/IDL-DataSets/

Simply go inside the folder `Dataset by Collection`,  download everything inside that folder.
Each `.csv` file contains all the filenames that a collection has. Extracting all filenames should be easy enough thing to do.

Then, read above.

**Comment by uakarsh** [comment]
I guess I need to understand and implement this. I would get back to you as soon as possible on this. I have downloaded the f.tar.gz.01, so maybe for that, I can figure out what are the PDFs for which OCRs are available.

Ultimately, this dataset would really be helpful for my learning and one of my research work.

Regards,
Akarsh

**Comment by Jordy-VL** [comment]
> @rubenpt91, can we have all the pdf names stored in json so that we can put it in our repo? I think it would be nice to have BOTH:
> 
> 1. all the pdfs names.
> 2. all the pdfs names which we have the OCR for.

Hi Ruben, this would indeed be nice :D

**Comment by furkanbiten** [comment]
Sorry guys, we were out of it for a while, you know life. I have talked with @rubenpt91 and hopefully, we can provide this. I do think it is quite important for the future of the data.

**Comment by Trott** [comment]
I'm not sure how big of an ask this is, but it would be great if someone volunteered to seed AcademicTorrents.com with the data. It would probably simplify things for folks. (It would also freeze the data in time. The IDL collection has new documents added all the time, so if someone downloads everything now, it's going to be different from what was used in this repo initially.)

**Comment by uakarsh** [comment]
Hi @furkanbiten and @Trott, hope you guys are well. I have worked on one of my task in which this dataset was involved, and this included downloading the images from the OCRs filename, so I downloaded one of the ocr folders i.e f.tar.gz, if I remembered it correctly, and most of the images was downloaded (corresponding to the OCRs) from the script I wrote. 

So, maybe I can help with something if required. And, for seeding data, I guess what we can do this, see what all images exist in the database, and then just create a text file containing the url address and the id. Does it sound good?

EDIT:

Since the data/image/pdf would be downloaded by taking the OCR's ID, it would not include the new additional files and would only keep those files, whose OCRs are available.

**Comment by furkanbiten** [comment]
@Trott, I absolutely agree and I wanted to do that when we first created the data. I think for me AcademicTorrents more than freezing the data, it is more convenient to download the data. However, the university had strict policies network-wise and the uni got hacked at some point so imagine me going to IT and asking for AcademicTorrents 😅  Yet, If anybody offers to seed the data, I think we can provide one-time access to all pdfs so that he does not have to wait for the download process (I still have to confim this but I believe we can).   

@uakarsh, thanks for offering to help. Hopefully, very soon, @rubenpt91 will provide the .txt for all the urls and names of the pdfs. He is currently working on it, but the bottleneck is the hard-drive. However, what I would suggest perhaps, either share the script or do a MR to the repo so that everyone can have access to the download script. 

So, If anybody offers to seed the data, please contact me and we can see what we can do about it.
@uakarsh, the download script would be nice to have.
Once @rubenpt91 has the .txt, I will write to notify you guys.

**Comment by uakarsh** [comment]
@furkanbiten It's been a while, since I have connected with you. Hope your health is good,and you are doing well.

I wanted to share the scripts (since they are the same I would be using to train one model I have worked on)

Here are the scripts:
https://github.com/uakarsh/docformer/blob/master/examples/docformer_pl/docformer_etoe_pretraining_finetuning/notebooks/1-docformer-pre-train-data-preparation.ipynb

Hope it helps. I have tried to include `joblib` since, it makes the workflow faster. And, the only thing needed currently is the `.tar.gz` files, and we are good to go with the dataset preparation.

**Comment by jiseo** [comment]
@uakarsh Hi! Thanks for your works. That's exactly what I find. I am currently working on a project about document understanding and tried to get IDL raw documents images. 
When I followed your scripts, I can download raw pdfs from url on web browser, but I couldn't get pdfs through python scripts. I got 403 error from requests. How did you solve this problem?

**Comment by uakarsh** [comment]
Hi @jiseo, did you try to print the url before fetching it? I probably once made a mistake about entering the correct URL, so if possible do check it once.

**Comment by jiseo** [comment]
@uakarsh Yes, I checked url and it works when i put it on chrome. But, it didn't work when I use requests library. I think the server blocked to access.

**Comment by uakarsh** [comment]
Probably, you can reduce the number of calls?

**Comment by jiseo** [comment]
Actually, at the first trial, I've got 403 error.

**Comment by uakarsh** [comment]
Hey @jiseo, apologies for the late reply. I just tried to run my notebook for downloading the PDF and then extracting the OCRs. It is still working in kaggle. You can try it out yourself: https://www.kaggle.com/code/akarshu121/1-docformer-pre-train-data-preparation?scriptVersionId=136923936

Let me know, if you still face the issue.
